### PR TITLE
Ensure the Patient resource type prefix in ProcedureRequest's subject…

### PR DESCRIPTION
**Pull request by sliver007**
… reference.

### CODE REVIEW

File: dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java [hunk 1]
```diff
@@ -147,7 +147,11 @@ private ProcedureRequest resolveProcedureRequest(ActivityDefinition activityDefi
         ProcedureRequest procedureRequest = new ProcedureRequest();
         procedureRequest.setStatus(ProcedureRequest.ProcedureRequestStatus.DRAFT);
         procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.ORDER);
-        procedureRequest.setSubject(new Reference(patientId));
+        String patientReferenceString = patientId;
+        if (!patientId.startsWith("Patient/")) {
+            patientReferenceString = "Patient/" + patientId;
```
Comment (brynrhodes) [inline dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java:152]: Reasonable, but what about if patientId is a contained reference? (e.g. `#1234`). And it should probably also account for the possibility that the reference is an absolute URL (e.g. `http://blah-be-de-blah...` and `uuid:oid:XXX...`)

Comment (sliver007) [reply dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java:155]: Used URI class to account for absoluteUrl and references to contained resources (indicated by presence of a fragment).

Comment (brynrhodes) [reply dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java:155]: Okay, but it does still need to check for the case that the input is relative, but already starts with `Patient/`

Comment (sliver007) [reply dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java:157]: Restored that check and also updated R4 with these same changes.

Comment (brynrhodes) [reply dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java:157]: Looks good

**Review by brynrhodes — APPROVED** [review]

**Pull request merged**